### DISCUSSION TITLE
Getting changed by someone else (or being given a diaper)

### DISCUSSION THREAD

**Discussion by TikkuTokku** [zippity21/Stardew_Valley_Regression_Mod#101] [Ideas]
I know there's a simple code in vanilla SDV for when you talk to someone so they can give you an item. All you need is the item code ID. For example, if you add [395] into the text a coffee will be added to your inventory. Do diapers have item code IDs? Can I get a list of them so I can have your spouse (or other villagers like Jodi who already gave you diapers at the beginning)) give you a diaper to change into? 

And similarly, is there an easy piece of code for changing out of one diaper and into another? If so, I would love the opportunity to have a dialogue box that makes your spouse change you into a new diaper for you.

And as I type this, is there a similarly easy code for turning a diaper into a wetter diaper? Because it would be particularly exciting for certain "meaner" characters to choose to use your diaper, or change you into an already wet one.

**Comment by zippity21** [comment]
The diapers, etc. that are added by this mod are not currently assigned IDs, but I was looking into using the content patcher (since its already a dependency) to rework how they are handled, so they can have IDs.

~~Forgive my ignorance, but what code let's an NPC dialogue give the player an item? I know that it's relatively easy in the mail system.~~ Neverminded, I see it as a special character.

For changing the Farmer, the code is relatively simple; it's a call to `ChangeUnderwear` to the Body object. Though getting an NPC to call it may not be so straight forward. Without looking too deep into it, I'd imagine you'd have to create a new event of some sort (also via content patcher).

The easiest way to trigger wetting is to call `Wet` on the body object. The downside is that you completely empty your bladder, as opposed to wetting a specific amount. If you wanted to create a separate function to add a specific amount, just use the `Wet` function as reference on how to handle leaks/blowouts. Again, to get an NPC to do this probably requires an event.

**Reply by Gooberology** [reply]
If the dialogue for being changed is handled by a function similar to the HandleVillager function for diaper usage reactions, you can run code alongside dialogue simultaneously. It is definitely something i want to see if i can add, as a complentary feature to being able to check other ABDL related characters.

**Reply by TikkuTokku** [reply]
That would open up so many creative opportunities for interacting with the villagers!!! 

My only real coding experience is with content patcher, so I really don't know how to do that. I don't even know how to open up the code if it's not a notepad file like the en document.

Also, I put in a pull request for Harvey, but I found a type since then, so give me a second and I'm going to try again.

**Reply by Gooberology** [reply]
I need to wrap mine up and stop trying to add so many features lol. I keep going at length about what im adding but then have trouble finding the time and never delivering :P

**Comment by TikkuTokku** [comment]
Thanks for getting back! If you do manage to give diapers an ID please let me know because that is blindingly easy to stick into dialogue. I have absolutely no idea how to trigger an event though. If that's an idea you think would be as exciting as I do I wouldn't mind trying to learn with some help.

**Comment by GardensStarlit** [comment]
Any shot of these features getting added at some point? I would love the ability to make NPCs give you a change or at least a clean diaper, and the idea of being able to manipulate (see: increase) your wetness/messiness without changing your farmer's bladder/bowels would be very neat

**Reply by Nillic** [reply]
I've looked into how much this is possible as some, and it's difficult.
Diapers in this mod are a specific item that doesn't work like other item do so there's no obvious to just so there's no obvious way to make the game use them as part of an event, for exemple. Underwear actually doesn't exist as items like clothing do and even if they did I don't think this game have a way to force change of clothing. There's a "do action" that simulate player clicking somewhere though, so if changing was linked to physical location instead of custom item, a change event would be possible maybe ? This also kinda make sense that diaper changing would be location specific to begin with.
I tried just using to call ChangeUnderwear with content code but couldn't make it because it's not from the base game and this mod can't use modded stuff like that, outside maybe some C# wizzardry I don't know about.
And even outside of that, there's also no way to track how your underwear statut either. This could change with all the new changes that now have queries and should be  more flexible but it's still probably a lot of work. But anything that allow to keep track of regression stuff (underwear types and states mostly).

But long story short mod is old and somewhat hard to work with when you want to had new stuff to it but it's core system is too solid, so restarting from scratches is not really attractive either. So keep expectations low or try to see how much you can work on it yourself.

**Reply by GardensStarlit** [reply]
Yeah, it's rough that the mod in that limbo between its dated code, and the time & care put into getting it where it's at now -- this is honestly one of the best ABDL mods I've run across, so I won't be looking a gift horse in the mouth; rest assured, I have no _expectations_ for anything more than we have right now, just _hopes_.

  ...Unfortunately, my _own_ coding knowledge is near-zero right now -- I'm more of a mod _user_, not a mod **_maker_** -- so, trying to jury-rig a solution into the existing code or building up a "version 2.0" myself are both a _bit_ above my grade, I'd say. The most I've done personally is homebrewing up some custom dialogue for Leah; which is **extremely** entry-level - more writing than coding. 
  (However, I will be taking a proper Computer Science course over the summer -- so, maybe I'll try and delve into C# as I go along, daunting as that sounds. Remaking the mod _would_ be a worthy pursuit if nobody's gone ahead and done it by the time I make it out of the learning-to-crawl phase of CS.)

**Comment by Gooberology** [comment]
> I've looked into how much this is possible as some, and it's difficult. Diapers in this mod are a specific item that doesn't work like other item do so there's no obvious to just so there's no obvious way to make the game use them as part of an event, for exemple. Underwear actually doesn't exist as items like clothing do and even if they did I don't think this game have a way to force change of clothing. There's a "do action" that simulate player clicking somewhere though, so if changing was linked to physical location instead of custom item, a change event would be possible maybe ? This also kinda make sense that diaper changing would be location specific to begin with. I tried just using to call ChangeUnderwear with content code but couldn't make it because it's not from the base game and this mod can't use modded stuff like that, outside maybe some C# wizzardry I don't know about. And even outside of that, there's also no way to track how your underwear statut either. This could change with all the new changes that now have queries and should be more flexible but it's still probably a lot of work. But anything that allow to keep track of regression stuff (underwear types and states mostly).
> 
> But long story short mod is old and somewhat hard to work with when you want to had new stuff to it but it's core system is too solid, so restarting from scratches is not really attractive either. So keep expectations low or try to see how much you can work on it yourself.

Ive already implemented it in my [1.5 fork](https://github.com/Gooberology/Stardew_Valley_Regression_Mod). I used a custom c# script and a json file to create a parser for custom dialogue command tags. I debugged a lot of it with Tikku, and he wrote some lines but the dialogue implementation is only partially finished.

**Reply by GardensStarlit** [reply]
Oh, hell yeah! 

I'll take a look through it and see how much I can wrap my head around, ((As I said in my reply to Nillic; I'm more of a dabbler than a proper coder)), But that's exciting to hear!

**Reply by Nillic** [reply]
Oh, that sound interesting. I guess I'll wait for this to be completed to 

This is specifically for dialogue ? When I was playing arround with the mod I did a bit of simple sprite edits for diapers but I'm not sure I've ever found a way to make them match current underwear.

**Reply by TikkuTokku** [reply]
It was a fun chunk of time workin' through all the bugs with Goober!
